packages/multi-test/src/wasm.rs

```diff
@@ -512,7 +512,12 @@ where
             let subres =
                 self.execute_submsg(api, router, storage, block, contract.clone(), resend)?;
             events.extend_from_slice(&subres.events);
-            Ok::<_, String>(subres.data.or(data))
+
+            if data.is_some() {
```

Comment:
This may cover the test cases, but is not the general solution we need.
It still makes no distinction between data returned in the sub message execute and in the reply processing submessage execute

Reply:
It does, just not in this very place. `execute_submsg` is modified, so it doesn't return data at all if reply was not executed, as in such a case those data are never to be used. At some point I even added checking for `ReplyOn` here, but when went deeper in this I realized it doesn't make sense - the logic behind deciding if reply is to be executed doesn't belong here, instead if there is no reply, then what is visible from `execute_submsg` should actually match what we expect - no data is set (as in such a case also wasmd implementation does not process this result in any way).